### PR TITLE
Tmap

### PR DESCRIPTION
**Pull request by JonSteinn**

### CODE REVIEW

File: tilings/tiling.py [hunk 1]
```diff
@@ -345,6 +346,128 @@ def _clean_assumptions(self) -> None:
...
+        by all.
+        """
+        gps = set(gps)
+        c, r, longest = cls._get_dimensions(gps)
+        max_len = longest if max_len == -1 else min(max_len, longest)
+        obs = cls._search(deque([GriddedPerm()]), gps, c, r, max_len)
+        return cls(obstructions=obs, requirements=(), assumptions=())
+
+    @staticmethod
+    def _extend_gp(gp: GriddedPerm, c: int, r: int) -> Iterator[GriddedPerm]:
```
Comment (enadeau) [inline tilings/tiling.py:368]: This should probably be turned into a method of `GriddedPerm`

Comment (christianbean) [reply tilings/tiling.py:368]: A similar function `GriddedPerm.insert_specific_point` might simplify this

Comment (JonSteinn) [reply tilings/tiling.py]: Fixed

**Comment by JonSteinn** [comment]
> This is a bit to much code to add to the tiling file. Can we only keep the `guess_from_gridded_perms` method and move all the logic to an algorithm file?

Done

**Review by enadeau — APPROVED** [review]

**Pull request merged**